ONSdigital/babbage
# Feature/update external and internal dependencies

### What

Updated external and internal dependencies to latest minor version.
Flying saucer [core](https://mvnrepository.com/artifact/org.xhtmlrenderer/flying-saucer-core) and [pdf](https://mvnrepository.com/artifact/org.xhtmlrenderer/flying-saucer-pdf) dependencies have been updated to version 4.2.1

### How to review

I added a note about the changes here - scroll down to babbage paragraph: https://confluence.ons.gov.uk/display/DIS/Java+repositories

Be sure you have java 1.8 as the current jdk and maven is using it (`mvn --version`)

- Run:  `mvn package` 
No  errors should appear, the build should run ok and the test cases should run ok

- Run and render some content in the app

- Run `mvn versions:display-dependency-updates -DallowMajorUpdates=false`
No libraries should be mentioned here that need updating

- Run zebedee reader in zebedee project root: ./run-reader.sh
    - zabadee needs content to render
- Run dp-api-router in dp-api-router project root: ./make debug
- Run babbage in babbage project root: ./run.sh
- Run sixteen in sixteen project root (provides style sheets for babbage): ./make debug
  - browse with browser to for example: http://localhost:8080/economy/inflationandpriceindices/consumerpriceinflation/2015-04-14
  - confirm page should render with no babbage error message or default error page   
  
  
### Who can review

Any member of ONS with experience in zebedee, dp_content, sixteen and babbage
